controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -152,6 +152,12 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(ctx context.Context
 		return ctrl.Result{}, err
 	}
 
+	enactmentProgressing, err := r.isEnactmentProgressing(instance)
+	if err != nil {
+		log.Error(err, "Error getting enactment status")
+		return ctrl.Result{}, err
+	}
+
 	err = r.initializeEnactment(*instance)
```

Comment:
Instead of re-read it before this, we can just return here the previousCondition as a pointer so we do something like 

```suggestion
	previousConditions, err = r.initializeEnactment(*instance)
	...
	if isEnactmentProgressing(previousConditions){...
```